appsec/src/extension/msgpack_helpers.c

```diff
@@ -14,9 +14,14 @@
 #include "php_helpers.h"
 #include "php_objects.h"
 
-static const int MAX_DEPTH = 32;
+static const size_t MAX_DEPTH_READING = 32;
+static const size_t MAX_DEPTH_WRITING = 20;
+static const size_t MAX_STR_LEN = 4096;
+static const size_t MAX_ARRAY_SIZE = 256;
```

Comment:
I know these limits are the "traditional" ones, but they don't make sense. Each array can have 256 elements, each element may be another array of 256 elements and so... 256^20 is a very large number. We should be limiting the size of the payload, or at the very least the number of elements

Reply:
You're right but this PR is moving from 0 limits to actually having limits so it's a net improvement in that respect. On the other hand, 256^20 would make PHP sweat long before even reaching this code. Generally I don't disagree with a better limit but discussions are moving towards a byte limit rather than a node limit (which is non-trivial to compute, but not impossible, but ultimately our limits are poor UX if they're ultimately exposed), so I'd suggest not implementing something that will ultimately deviate from every other implementation.